CommunityFridgeMapApi/functions/fridges/v1/app.py

```diff
@@ -0,0 +1,54 @@
+import os
+import logging
+import json
+
+try:
+    from db import get_ddb_connection, Fridge
+except:
+    # If it gets here it's because we are performing a unit test. It's a common error when using lambda layers
+    # Here is an example of someone having a similar issue:
+    # https://stackoverflow.com/questions/69592094/pytest-failing-in-aws-sam-project-due-to-modulenotfounderror
+    from dependencies.python.db import get_ddb_connection, Fridge
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class FridgHandler:
+    @staticmethod
+    def get_fridge_id(event: dict):
+        """
+        Gets the fridge_id from the event dictionary
+        """
+        fridge_id = None
+        if event["pathParameters"] is not None:
+            fridge_id = event["pathParameters"].get("fridge_id", None)
+        return fridge_id
+
+    @staticmethod
+    def lambda_handler(event: dict, ddbclient: "botocore.client.DynamoDB") -> dict:
+        """
+        Extracts the necessary data from events dict, and executes a function corresponding
+        to the event httpMethod
+        """
+        httpMethod = event.get("httpMethod", None)
+        fridge_id = FridgHandler.get_fridge_id(event)
+        db_response = None
+        if httpMethod == "GET":
+            if fridge_id:
+                db_response = Fridge(db_client=ddbclient).get_item(fridge_id)
+            else:
+                raise ValueError(f"Query Fridges API Not Yet Developed")
+        elif httpMethod == "POST":
+            pass
+        else:
+            raise ValueError(f'Invalid httpMethod "{httpMethod}"')
+        return db_response.api_format()
+
+
+def lambda_handler(
+    event: dict, context: "awslambdaric.lambda_context.LambdaContext"
+) -> dict:
+    env = os.environ["Environment"]
+    ddbclient = get_ddb_connection(env)
+    return FridgHandler.lambda_handler(event, ddbclient)
```

Comment:
#### Question
This might be a stupid question, but is it recommended to put `lambda_handler` and `get_fridge_id` in `FridgeHandler` class instead of defining them at module-level? I'm not quite familiar with serverless function conventions :sweat_smile:

Reply:
Looks good to me!

Reply:
No stupid questions haha :)!

Not sure if it's recommended or not. I generally put related functions under the same class. It's not a convention that is tied to serverless functions (lambdas). The only thing that you actually need in your lambdas are this function:

```
def lambda_handler(
    event: dict, context: "awslambdaric.lambda_context.LambdaContext"
) -> dict:
```

Reply:
Honestly could've probably just put everything under the lambda_handler function instead of having a separate class. Was just trying to keep the same structure as some of the first APIs that we wrote

Reply:
Yes, it make sense. Thanks!

Reply:
Sorry for the spam. Was just thinking through why we chose to do it from the beginning. 

It helps with writing tests. 

Our `FridgReportHandler.lambda_handler()` takes in a `ddbclient: "botocore.client.DynamoDB"` which we can mock for unit tests

Reply:
Ohh that's actually clever :+1: